lcls_tools/common/data/emittance/fit.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+from scipy.optimize import minimize
+import importlib.util
+from matrix_calcs import propagate_twiss, twiss_transport_mat_from_rmat, build_quad_rmat
+from lcls_tools.common.data.model_general_calcs import bmag_func
+
+
+def compute_emit_bmag(
+    k: np.ndarray,
```

Comment:
Can we use Pydantic to do some type checking here?

Reply:
Do you want me to make an emittance class (pydantic object) with this function as a class method instead of a standalone function?

Reply:
Since we're renaming the file to emittance.py, let's hold off on that so you can use this next week as is. Can you write another issue for us to think about reformatting? I don't want a user to have to `import emittance.Emittance`, so we should think about it a little more.

Reply:
After discussion, let's write an issue on the subject of making this a class or not. Also remove the 'emittance' folder, and put emittance.py in data folder.